lib/board.rb

```diff
@@ -1,16 +1,142 @@
 class Board
-  attr_reader :cells
+  attr_reader :cells, :game_checker
 
-  def initialize
+  def initialize(game_checker)
     @cells = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    @game_checker = game_checker
   end
 
   def find_position(num)
     @cells[num - 1]
   end
 
   def place_mark(mark, position)
-    pos = position.to_i - 1
-    @cells[pos] = mark
+    cell_location = position.to_i - 1
+    @cells[cell_location] = mark
+  end
+
+  def won?(player1, player2)
+    @game_checker.winner?(self, player1) || @game_checker.winner?(self, player2)
+  end
+
+  def tie?(player1, player2)
+    @game_checker.tie?(self, player1, player2)
+  end
+
+  def full?
+    @cells.each_with_index do |cell, idx|
+      if cell.to_i === idx + 1
+        return false
+      end
+    end
+    true
+  end
+
+  def row_full?(mark)
+    rows = find_rows
+    rows.each do |row|
+      if row.all?(mark)
+        return true
+      end
+    end
+    false
+  end
+
+  def col_full?(mark)
+    cols = find_cols
+    cols.each do |col|
+      if col.all?(mark)
+        return true
+      end
+    end
+    false
+  end
+
+  def diagonal_full?(mark)
+    diags = find_diagonals
+    diags.each do |diag|
+      if diag.all?(mark)
+        return true
+      end
+    end
+    false
+  end
+
+  def fill_row(mark, row_num)
```

Comment:
These methods are for testing?  No need to include in the board, can just include in the test itself.  For example, in the test you would have the following method:
def fill_row(mark)
```
board.place_mark(1, mark)
board.place_mark(2, mark)
board.place_mark(3, mark)
end
```

And then you can call this method, instead of calling the board's place_mark method within the it block.

Reply:
OH that makes a lot more sense!